packages/filesystem/src/browser/file-tree/file-tree.ts

```diff
@@ -28,6 +28,30 @@ export class FileTree extends TreeImpl {
     @inject(FileSystem) protected readonly fileSystem: FileSystem;
     @inject(LabelProvider) protected readonly labelProvider: LabelProvider;
 
+    @postConstruct()
+    protected init(): void {
+        this.toDispose.push(
+            this.labelProvider.onElementUpdated((uri: URI) => this.doUpdateElement(uri))
+        );
+    }
+
+    protected doUpdateElement(uri: URI): void {
```

Comment:
Have you tested this change with simple and multi-root workspace? Please add tests to `how to test` section for others. You could also push one test commit with the test contribution instead of explaining how to create it, just don't sign it that we don't forget to remove it later.